docs/markdown/Reference-manual.md

```diff
@@ -1825,6 +1825,36 @@ which to build the targets.
 If you desire more specific behavior than what this command provides,
 you should use `custom_target`.
 
+### range()
+
+``` meson
+    rangeobject range(stop)
+    rangeobject range(start, stop[, step])
+```
+
+*Since 0.57.0*
+
+Return an opaque object that can be only be used in `foreach` statements.
+- `start` must be integer greater or equal to 0. Defaults to 0.
+- `stop` must be integer greater or equal to `start`.
+- `step` must be integer greater or equal to 1. Defaults to 1.
+
+It cause the `foreach` loop to be called with the value from `start` included
+to `stop` excluded with an increment of `step` after each loop.
+
+```meson
+# Loop 15 times with i from 0 to 14 included.
+foreach i : range(15)
```

Comment:
Is there a particular reason or precedent elsewhere that made you go for a `[start, stop[` range rather than an inclusive `[start, stop]`? I personally find the latter more intuitive.

Reply:
That's what Python does. Can be seen both ways, range(5) does 5 loops, from 0 to 4...

Reply:
Same thing than common c loop for(i=0; I < stop; i++)

Reply:
https://en.m.wikipedia.org/wiki/Off-by-one_error